### ISSUE TITLE
Can't load PvPGN server with 1.27b (1.27.1.7085)

### ISSUE THREAD

**Issue by shiioo** [w3lh/w3l#9]
Using the w3l exe/dll -1.5.1.1 SRP3 version- built by Keres (found on the Wiki).
Game doesn't show the PvPGN server registered and instead displays the list of official BNet servers.
Patcher seems to work as I can load the game in windowed mode.

Base games are the v1.27 (installers provided by Blizzard), then patched to v1.27b.
All that running on Win7 x64.

edit: I also compiled these 3 files myself, same result.

**Comment by cen1** [comment]
w3l does not modify gateways in any way afaik. If your w3 loads, it works. YOu need to add your own pvpgn servers via regedit.

**Comment by shiioo** [comment]
It does not modify them indeed, but I added one using this: https://github.com/pvpgn/battle.net-gateway-installer
Even the old gateway editor detects the change, but for some reason only the official ones are displayed.

**Comment by cen1** [comment]
Just edit Battle.net Gateways registry entry and add on the bottom
```
my pvpgn ip
-1
gateway name
```

**Comment by shiioo** [comment]
There's just one entry there, and it's mine already :)
It's 1 instead of -1 though, but adding the minus doesn't change anything.

**Comment by cen1** [comment]
Maybe the fact that you deleted official gateways is the problem. Try having official plus your own.

**Comment by shiioo** [comment]
There's no `Battle.net Gateways` entry at all upon doing a clean installation, thus nothing to delete.
It's created after using that batch file.

Maybe it has to do with the fact that the base game and its expansion (both v1.27) were retrieved through the Blizzard downloader. I then manually patched them to v1.27b.

**Comment by w3lh** [comment]
cen1 is right, loader has no any connection to gateways list. Blizzard's installer resets gateway list to the default. You have to modify gateway list by yourself.

**Comment by shiioo** [comment]
That's what I did.
Upon a clean installation, that registry entry isn't there at all.
I used the batch file provided on the PvPGN github to create it and add a custom server.
Result is that it doesn't work and only the official servers are listed.

Though I found the issue, at least for me.
I used my old .reg file and this one works.

Instead of having:
`ip
1
name`

It has:
`
1001
00
ip
1
name
`

But you were indeed right, issue wasn't with the loader.
Sorry for the trouble!
